submissions/eoneoff/data-structures-http/server/handler.js

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable default-case */
+/* eslint-disable no-return-assign */
+/* eslint-disable func-names */
+
+function isValidType(value) {
+  return typeof value === 'string' || typeof value === 'number';
+}
+
+function isJson(request) {
+  return request.headers['content-type'] === 'application/json';
+}
+
+function postStack(response, stack, data) {
+  stack.push(data.data);
+  response.writeHead(200);
+  response.end();
+}
+
+function deleteStack(response, stack) {
+  response.writeHead(200);
+  response.end(JSON.stringify({ data: stack.pop() }), {
+    'Content-Type': 'application/json'
+  });
+}
+
+function getList(response, list) {
+  response.writeHead(200, { 'Content-Type': 'application/json' });
+  response.end(JSON.stringify({ data: list.showList() }));
+}
+
+function postList(response, list, data) {
+  if (!data.successor || isValidType(data.successor)) {
+    try {
+      list.insert(data.data, data.successor);
+      response.writeHead(200);
+    } catch (err) {
+      response.writeHead(400, err.message);
+    }
+  } else {
+    response.writeHead(400, 'Wrong data type');
+  }
+  response.end();
+}
+
+function deleteList(response, list, data) {
+  try {
+    list.remove(data.data);
+    response.writeHead(200);
+  } catch (err) {
+    response.writeHead(400, err.message);
+  } finally {
+    response.end();
+  }
+}
+
+function notFound(response) {
+  response.writeHead(404, 'Url not found');
+  response.end();
+}
+
+function wrongDataType(response) {
+  response.writeHead(400, 'Wrong data type');
+  response.end();
+}
+
+function wrongContentType(response) {
+  response.writeHead(400, 'Wrong content type');
+  response.end();
+}
+
+function parseJsonRequest(request, response, stack, list, data, structureType) {

```

Comment:
I don't like this function too. But the problem is that is is situated in a callback and heavily depends on the results, returned by previous callbacks and closures of the handler function. So if I separate it, It will make a callback in handler more complicated and it is quite big already.